feat: Generate `UserProductList` when we detect product intent

rafaeelaudibert opened the pull request:
One of the cases where we wanna automatically generate a new `UserProductList` is when we
detect product intent - which teams already trigger from the UI.

This just hooks onto `ProductIntent.register` and creates the item in the sidebar. The UI
is also automatically refreshing to show it on the sidebar - properly respecting user preferences, ofc

[Screen Recording 2025-11-20 at 19.36.12.mov <span class="graphite__hidden">(uploaded via Graphite)</span> <img class="graphite__hidden" src="https://app.graphite.com/user-attachments/thumbnails/f373d264-f231-48d0-90ec-bbb7c0e34d6c.mov" />](https://app.graphite.com/user-attachments/video/f373d264-f231-48d0-90ec-bbb7c0e34d6c.mov)

rafaeelaudibert committed:
feat: Generate `UserProductList` when we detect product intent

One of the cases where we wanna automatically generate a new `UserProductList` is when we
detect product intent - which teams already trigger from the UI.

This just hooks onto `ProductIntent.register` and creates the item in the sidebar. The UI
is also automatically refreshing to show it on the sidebar - properly respecting user preferences, ofc.

rafaeelaudibert committed:
feat: Add custom `useLocalStorage` hook

This simplifies using a value from the local storage as a hook

rafaeelaudibert committed:
feat: Improve tips on sidebar

Sidebar tips now keep displaying for slightly longer and can be dismissed.

MattBro reviewed (approved):
Looks good, I think the CI/greptile comments are valid, not worth blocking the PR over, but would be good enhancements.

rafaeelaudibert committed:
feat: Configure onboarding product intent

We're now properly adding the products to the sidebar on onboarding with the correct reason on the `UserProductList` model

rafaeelaudibert committed:
fix: Fix type

rafaeelaudibert committed:
fix: Fix mypy and broken API validations

rafaeelaudibert committed:
fix: fix types in tests
